src/componententity.cpp

```diff
@@ -349,15 +250,21 @@ UnitsPtr ComponentEntity::takeUnits(const std::string &name)
     return takeUnits(mPimpl->findUnits(name) - mPimpl->mUnits.begin());
 }
 
-void ComponentEntity::replaceUnits(size_t index, const UnitsPtr &units)
+bool ComponentEntity::replaceUnits(size_t index, const UnitsPtr &units)
 {
-    removeUnits(index);
-    mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);
+    bool status = false;
+    if (index < mPimpl->mUnits.size()) {
+        removeUnits(index);
+        mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);
+        status = true;
+    }
+
+    return status;
 }
```

Comment:
Why not simply have the following since `removeUnits()` now returns a boolean?

```
bool ComponentEntity::replaceUnits(size_t index, const UnitsPtr &units)
{
    bool status = false;

    if (removeUnits(index)) {
        mPimpl->mUnits.insert(mPimpl->mUnits.begin() + index, units);

        status = true;
    }

    return status;
}
```